### PR TITLE
feat(ulimits): additional ulimit options

### PR DESCRIPTION
**Pull request by chrisminton**
## Description

- add as, cpu, maxlogins, msgqueue, rss, sigpending options to the `user_ulimit` resource
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Review by Daemonali — APPROVED** [review]

**Comment by johnmccrae** [comment]
Can you add some tests in for these new options please?

**Comment by chrisminton** [comment]
Hi @johnmccrae there are additions for the `kitchen-tests` section - this is currently the only place that ulimit options are being interrogated. Is there something you'd like me to add?

**Comment by tpowell-progress** [comment]
> Hi @johnmccrae there are additions for the `kitchen-tests` section - this is currently the only place that ulimit options are being interrogated. Is there something you'd like me to add?

@chrisminton Thanks for pointing those out. I think we're good with those.

File: lib/chef/resource/user_ulimit.rb [hunk 1]
```diff
@@ -85,19 +104,38 @@ class UserUlimit < Chef::Resource
...
             process_soft_limit: new_resource.process_soft_limit,
             process_hard_limit: new_resource.process_hard_limit,
+            locks_limit: new_resource.locks_limit,
             memory_limit: new_resource.memory_limit,
+            maxlogins_limit: new_resource.maxlogins_limit,
+            maxlogins_soft_limit: new_resource.maxlogins_soft_limit,
+            maxlogins_hard_limit: new_resource.maxlogins_hard_limit,
+            msgqueue_limit: new_resource.msgqueue_limit,
+            msgqueue_soft_limit: new_resource.msgqueue_soft_limit,
+            msgqueue_hard_limit: new_resource.msgqueue_hard_limit,
```
Comment (jaymzh) [inline lib/chef/resource/user_ulimit.rb:123]: this isn't in the template at all.

**Review by jaymzh — CHANGES_REQUESTED** [review]
one small bug.

File: lib/chef/resource/user_ulimit.rb (hunk 1 above)
Comment (chrisminton) [reply lib/chef/resource/user_ulimit.rb:123]: fixed thanks @jaymzh

**Review by jaymzh — CHANGES_REQUESTED** [review]

**Comment by johnmccrae** [comment]
@chrisminton Can you add the test that Phil requested?

File: kitchen-tests/cookbooks/end_to_end/recipes/linux.rb [hunk 2]
```diff
@@ -93,16 +93,27 @@
...
+  as_hard_limit "unlimited"
+  cpu_soft_limit 1024
+  cpu_hard_limit 8096
   filehandle_soft_limit 8192
   filehandle_hard_limit 8192
   process_soft_limit 61504
   process_hard_limit 61504
+  locks_limit 8192
+  maxlogins_soft_limit 5
+  maxlogins_hard_limit 10
```
[the remark this answers is not in the document]
Comment (chrisminton) [reply kitchen-tests/cookbooks/end_to_end/recipes/linux.rb:106]: done @jaymzh

**Review by tpowell-progress — APPROVED** [review]

**Comment by tpowell-progress** [comment]
@chrisminton Looks like the Buildkite build is broken due to an old configuration file. Can you rebase on main, please?

**Comment by tpowell-progress** [comment]
Approving workflows and tagging for community PR review.

**Pull request merged**